pkg/commands/git_commands/rebase.go

```diff
@@ -137,7 +137,7 @@ func (self *RebaseCommands) PrepareInteractiveRebaseCommand(baseShaOrRoot string
 		debug = "TRUE"
 	}
 
-	cmdStr := fmt.Sprintf("git rebase --interactive --autostash --keep-empty --no-autosquash %s", baseShaOrRoot)
+	cmdStr := fmt.Sprintf("git rebase --interactive --autostash --keep-empty --empty=keep --no-autosquash %s", baseShaOrRoot)
```

Comment:
Hey! Can I ask if this change was necessary to fix the bug you were fixing? Or was it a separate/unrelated change? I've noticed that my commits are no longer dropped when they become empty after rebasing. I would like to go back to the previous behaviour.

Reply:
No, it wasn't really necessary for fixing the bug. I added this to make the behavior consistent between various patch operations. For example, when creating a patch for an entire commit, and then using either "move patch into index", or "move patch into new commit", then the original commit becomes empty. This has nothing to do with `--empty=keep`, it is because we do an interactive rebase that stops at the source commit, apply the patch in reverse, and amend the source commit using `git commit --amend --allow-empty`. We need to pass `--allow-empty` here, otherwise the commit would fail with an error. It has behaved like that before this PR.

For moving a patch into a different commit, the behavior was different depending on whether the source commit would come before or after the destination commit. If the source commit was older, it would become empty, for the same reason as above. However, if the source commit would come later in the history, then it would be dropped; that's why I added the `--empty=keep`.

With all that said, I'm not sure either if I like the behavior. For situations other than moving patches between commits, it's surprising when commits become empty and you have to drop them manually. Maybe it would be better to live with the inconsistency and remove the `--empty=keep` again.

@jesseduffield Do you have an opinion about this?

Reply:
OK, I have run into this enough times myself now that I'm sure it was a mistake to add `--empty=keep` unconditionally to all our rebases.

My proposed solution is to add `--empty=keep` only for the specific case of moving a patch to an earlier commit, to keep the behavior of this consistent with moving the patch to a later commit; but omit it for all other cases. Alternatively, we could drop it entirely and live with the inconsistency of moving a patch up vs. down.

@jesseduffield Thoughts?

Reply:
Sorry for missing your last comment. I agree, let's only apply the flag when moving commits up/down

Reply:
PR is here: #2644